packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -126,16 +126,28 @@ function withoutLocalOpMetadata(message: IPendingMessage): IPendingMessage {
 }
 
 /**
- * Get the effective batch ID for a pending message.
- * If the batch ID is already present in the message's op metadata, return it.
+ * Get the effective batch ID for the input argument.
+ * Supports either an IPendingMessage or an InboundBatch.
+ * If the batch ID is explicitly present, return it.
  * Otherwise, generate a new batch ID using the client ID and batch start CSN.
- * @param pendingMessage - The pending message
- * @returns The effective batch ID
  */
-function getEffectiveBatchId(pendingMessage: IPendingMessage): string {
+function getEffectiveBatchId(
+	pendingMessageOrInboundBatch: IPendingMessage | InboundBatch,
```

Comment:
What's the best practice when wanting to do the same action for different types? I find this function a bit odd, expecting two different types and having an if to basically determine which type is but it may actually be the best practice, instead of two different functions or other pattern